docs/going-to-production.md

```diff
@@ -0,0 +1,128 @@
+  </ul>
+</details>
+
+Caching improves response times and reduces the number of requests to external services. Next.js automatically adds caching headers to immutable assets served from `/_next/static` including JavaScript, CSS, static images, and other media.
+
+```
+Cache-Control: public, max-age=31536000, immutable
+```
+
+`Cache-Control` headers set in `next.config.js` will be overwritten in production to ensure that static assets can be cached effectively. If you need to revalidate the cache of a page that has been [statically generated](/docs/basic-features/pages.md#static-generation-recommended), you can do so by setting `revalidate` in the page's [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) function. If you're using `next/image`, there are also [specific caching rules](/docs/basic-features/image-optimization.md#caching) for the default Image Optimization loader.
+
+**Note:** When running your application locally with `next dev`, your headers are overwritten to prevent caching locally.
+
+```
+Cache-Control: no-cache, no-store, max-age=0, must-revalidate
+```
+
+You can also use caching headers inside `getServerSideProps` and API Routes for dynamic responses. For example, using [`stale-while-revalidate`](https://web.dev/stale-while-revalidate/).
+
+```jsx
+// This value is considered fresh for ten seconds (s-maxage=10).
+// If a request is repeated within the next 10 seconds, the previously
+// cached value will still be fresh. If the request is repeated before 59 seconds,
+// the cached value will be stale but still render (stale-while-revalidate=59).
+//
+// In the background, a revalidation request will be made to populate the cache
+// with a fresh value. If you refresh the page, you will see the new value.
+export async function getServerSideProps({ req, res }) {
+  res.setHeader(
+    'Cache-Control',
+    'public, s-maxage=10, stale-while-revalidate=59'
+  )
+
+  return {
+    props: {},
+  }
+}
+```
+
+**Note:** Your deployment provider must support edge caching for dynamic responses. If you are self-hosting, you will need to add this logic to the edge yourself using a key/value store. If you are using Vercel, [edge caching works without configuration](https://vercel.com/docs/edge-network/caching).
+
+## Reducing JavaScript Size
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-dynamic-import">with-dynamic-import</a></li>
+  </ul>
+</details>
+
+To reduce the amount of JavaScript sent to the browser, you can use the following tools to understand what is included inside each JavaScript bundle:
+
+- [Import Cost](https://marketplace.visualstudio.com/items?itemName=wix.vscode-import-cost) – Display the size of the imported package inside VSCode.
+- [Package Phobia](https://packagephobia.com/) – Find the cost of adding a new dev dependency to your project.
```

Comment:
```suggestion
- [Package Phobia](https://packagephobia.com/) – Find the cost of adding a new dev dependency to your project.
- [Bundle Phobia](https://bundlephobia.com/) - Analyze how much a dependency can increase bundle sizes.
```

Do we wanna mention https://bundlephobia.com/ too?

Reply:
They do the same thing, right? Let's just use one.

Reply:
No, one analyzes the install size of the package and one analyzes how it contributes to your bundle.

Notice difference with analyzing react:

https://bundlephobia.com/package/react@17.0.2
https://packagephobia.com/result?p=react

Reply:
Next is blocked 😢 https://bundlephobia.com/package/next

Reply:
Makes sense to show that for a bundle analyzer since next isn't a client-side dependency right? The tool would have have to be aware of the client components that next exposes that impact client bundles.

Reply:
True that 👍